### DISCUSSION TITLE
Vercel: Application error: a client-side exception has occurred

### DISCUSSION THREAD

**Discussion by Irazhap** [neuefische/questions#611] [Web] [answered]
Help!!!
I have deployed my project on Vercel and get the following error message: Application error: a client-side exception has occurred (see the browser console for more information).

![Bildschirmfoto 2023-05-23 um 09 27 33](https://github.com/neuefische/questions/assets/92851229/4cc1a706-f885-4708-aa37-48d173dbf15a)

I get the problem on the pages where I reflect the api answer. Everything works locally for me.

All details about my code can be found in my pull request: [PR](https://github.com/Irazhap/capstone-project/pull/13)

**Comment by t0bbl** [comment]
`  return (
    <Container>
      {movies.map((movie) => (
        <List key={movie.id}>
          <CardImage
            src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
            alt={movie.title}
            width={140}
            height={210}
          />
        </List>
      ))}
    </Container>
  );`

i think if you add a ? after movies it should render when it has the data, not before:

`  return (
    <Container>
      {movies?.map((movie) => (
        <List key={movie.id}>
          <CardImage
            src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
            alt={movie.title}
            width={140}
            height={210}
          />
        </List>
      ))}
    </Container>
  );`

also did you add the Environment Variables in vercel for your api?

**Reply by Irazhap** [reply]
I tried, the problem persists

**Reply by onemanwenttomow** [reply, marked as answer]
I think @t0bbl has the right idea. The error message suggests that `movies` is undefined. 

``` js
TypeError: Cannot read properties of undefined (reading 'map')
```

Guarding against that with optional chaining as suggested would be a good option. 

``` js
{movies?.map((movie) => (
  <List key={movie.id}>
    <CardImage
      src={`https://image.tmdb.org/t/p/w500/${movie.poster_path}`}
      alt={movie.title}
      width={140}
      height={210}
    />
  </List>
))}
```

The movies on the home page seem to be coming from [here](https://github.com/Irazhap/capstone-project/blob/fetching/lib/db.js) (i.e. locally). 

We need to figure out why we are not getting the movies from your API routes. 

When i run your project locally and use my own API key i get the data no problem. If i remove the key i get the following response back

``` js 
data {
  status_code: 7,
  status_message: 'Invalid API key: You must be granted a valid key.',
  success: false
}
```

In your code you are then doing this 

``` js
export default async function fetcher(url) {
  const apiResponse = await fetch(url);
  const data = await apiResponse.json();
  return data.results;
}

```

where `data.results` will be `undefined`. 

Can you double check again that you have correctly added the API KEY to your next.js project? You might need to tell it to rebuild the project after adding the key. 



![Screenshot 2023-05-23 at 16 28 13](https://github.com/neuefische/questions/assets/1374487/26dddce6-9b09-4968-bf8c-e60bf7517a6d)

You can try and redeploy your branch as follows. (make sure you select the correct branch).

![Screenshot 2023-05-23 at 16 31 23](https://github.com/neuefische/questions/assets/1374487/a08b5b1a-f5b5-4765-ac05-b34ab19f6907)

**Reply by Irazhap** [reply]
@onemanwenttomow 
thanks, the problem was with the redeploy
